app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -0,0 +1,50 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    # Generic JWT authenticator that receive JWT vendor configuration and uses to validate that the authentication
+    # request is valid, and return conjur authn token accordingly
+    Authenticate = CommandClass.new(
+      dependencies: {
+        token_factory: TokenFactory.new
+      },
+      inputs: %i[jwt_configuration authenticator_input]
+    ) do
+
+      def call
+        validate_and_decode_token
+        get_conjur_id
+        validate_restrictions
+        new_token
+      end
+
+      private
+
+      def validate_and_decode_token
+        unless @jwt_configuration.validate_and_decode_token
+          raise "validate and decode failed"
+        end
+      end
+
+      def get_conjur_id
+        # Will be changed when real get identity implemented
+        unless @jwt_configuration.get_conjur_id == "cucumber"
+          raise "wrong identity"
+        end
+      end
+
+      def validate_restrictions
+        unless @jwt_configuration.validate_restrictions
```

Comment:
to validate restrictions you need to pass restrictions; where are they?
It feels the methods signature are not finalized.
The methods signature should be in the design, and the objects they need to get should be part of the sequential flow.